### ISSUE TITLE
Missing rows in large files

### ISSUE THREAD

**Issue by anvaka** [klaemo/csv-stream#14]
Hello,

I was trying to process large csv file (~100MB+), and found that this library does not properly return all rows.

Use this dataset to reproduce: [description.csv.gz](https://s3.amazonaws.com/csvstreamifydata/description.csv.gz)

Code:
``` js
// streamify_bug.js
var inputFile = require('fs').createReadStream('./description.csv'),
  csv = require("csv-streamify"),
  parser = csv({
    objectMode: true,
    columns: true
  });

parser.on('readable', printLine);

inputFile.pipe(parser);

function printLine() {
  var line = parser.read();
  console.log(parser.lineNo);
}
```

If I count number of produced lines (`node streamify_bug.js | wc -l`) I get `993,319`. But `description.csv` has:

```
> wc -l description.csv
1635938 description.csv
```

`1,635,938` lines.

I'm using `csv-streamify`  version 1.0.0. `npm version`:

```
{ http_parser: '1.0',
  node: '0.10.32',
  v8: '3.14.5.9',
  ares: '1.9.0-DEV',
  uv: '0.10.28',
  zlib: '1.2.3',
  modules: '11',
  openssl: '1.0.1i',
  npm: '1.4.28' }
```

**Comment by klaemo** [comment]
For the (almost) 1 year anniversary of this issue, I made us the present of a fix :D It emits all the 1.6 millions lines now.
This also fixes the memory leak I was seeing. A bug in the handling of quotes caused thousands of lines to be buffered in memory.

Thank you for the test data :) Once Travis passes I will release `3.0.2` with a fix.

**Comment by anvaka** [comment]
Thank you :)!
